backend/agent_logic/tools/sandbox_executor.py

```
@@ -1,79 +1,257 @@
             
+        except docker.errors.ContainerError as e:
+            logger.error(f"? Container execution error: {str(e)}")
+            return {
+                "success": False,
+                "output": "",
+                "error": str(e),
+                "exit_code": e.exit_status if hasattr(e, 'exit_status') else -1
+            }
+        
+        except docker.errors.ImageNotFound as e:
+            logger.error(f"? Docker image not found: {str(e)}")
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)